src/org/genepattern/server/webapp/rest/api/v1/job/GetPipelineJobLegacy.java

```diff
@@ -136,28 +147,103 @@ public JSONObject getJob(final GpContext userContext, final JobInfo jobInfo, fin
                 throw new GetJobException(errorMessage + ": "+t.getLocalizedMessage());
             }
         }
-        if (includePermissions && job!=null) {
-            //only include permissions for the top-level job
-            try {
-            JSONObject permissions=initPermissionsFromJob(userContext, jobInfo);
-            if (permissions!=null) {
-                job.put("permissions", permissions);
-            }
-            }
-            catch (Throwable t) {
-                final String errorMessage="Error initializing permissions for jobId="+jobInfo.getJobNumber();
-                log.error(errorMessage, t);
-                throw new GetJobException(errorMessage + ": "+t.getLocalizedMessage());
+        
+       return job;
+    
+    }
+    
+    protected static LoadingCache<String, JSONObject> jobCache;
+    protected static final HashMap<String, Object[]> paramMap = new HashMap<String, Object[]>();
+    
+    public JSONObject getJob(final GpContext userContext, final JobInfo jobInfo, final boolean includeChildren,
+                             final boolean includeOutputFiles, final boolean includePermissions,
+                             final boolean includeComments, final boolean includeTags) throws GetJobException {
+        //manually create a JSONObject representing the job
+        
+        final String composite_key = ""+jobInfo.getJobNumber() + includeChildren + includeOutputFiles + includeComments + includeTags;
+        Object[] params = new Object[6];
+        params[0]=jobInfo;
+        params[1]=includeChildren;
+        params[2]=includeOutputFiles;
+        params[3]=includeComments;
+        params[4]=includeTags;
+        params[5]=userContext;
+        paramMap.put(composite_key, params);
+        // ***** the paramMap is a hack to get all this stuff up via the composite key even though it must be final
+        if (jobCache == null){
+            jobCache = CacheBuilder.newBuilder()
+                .maximumSize(1000)
```

Comment:
should we parameterize the maximumSize? It's hard-coded to 1000. Typically there are 4000 active jobs on the server. What happens when we reach the max?